sql/hive/src/main/scala/org/apache/spark/sql/hive/hiveWriterContainers.scala

```diff
@@ -212,7 +212,7 @@ private[spark] class SparkHiveDynamicPartitionWriterContainer(
     }
 
     val dynamicPartPath = dynamicPartColNames
-      .zip(row.toSeq.takeRight(dynamicPartColNames.length))
+      .zip(row.toSeq(schema).takeRight(dynamicPartColNames.length))
```

Comment:
A test failed as the scema dismatches the row. I'm not familiar with the hive part, does someone have any idea on it?